### PR TITLE
[v1] Add urfave_cli_no_docs build tag

### PR DESCRIPTION
**Pull request by kolyshkin**
## What type of PR is this?

- feature

## What this PR does / why we need it:

Add urfave_cli_no_docs build tag

This removes the resulting binary dependency on cpuguy83/md2man and
russross/blackfriday (and a few more packages imported by those),
which saves more than 400 KB (more than 300 KB
once stripped) from the resulting binary.

## Which issue(s) this PR fixes:

none (although see some discussions in e.g. #1055)

## Special notes for your reviewer:

This is a backport of https://github.com/urfave/cli/pull/1375 to v1 branch.

## Testing

Unit tests with this tag set are added to GHA CI (see last commit).

## Release Notes

```release-note
Added support for `urfave_cli_no_docs` tag to reduce the binary size by 300+ KB
in case your application is not using `ToMarkdown` and `ToMan` methods.
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
Ah, it seems I have broken CI with an earlier PR. Fixed here.

**Comment by kolyshkin** [comment]
Draft until I fix v1 CI (in progress).

**Comment by kolyshkin** [comment]
Draft until https://github.com/urfave/cli/pull/1384 is merged.

**Comment by kolyshkin** [comment]
@meatballhat this one is ready to go!

**Comment by meatballhat** [comment]
@kolyshkin did you want to take care of taking the PR out of Draft mode and editing out the `[DO NOT MERGE]` before I do stuff? I'm happy to take care of those bits, too.

**Comment by kolyshkin** [comment]
> @kolyshkin did you want to take care of taking the PR out of Draft mode and editing out the `[DO NOT MERGE]` before I do stuff? I'm happy to take care of those bits, too.

Sorry, forgot to remove `[DO NOT MERGE]` yesterday. Done.

Can you please make a release once this is merged?

**Review by meatballhat — APPROVED** [review]

**Pull request merged**
